crates/precompile/src/point_evaluation.rs

```diff
@@ -0,0 +1,55 @@
+use crate::{Precompile, PrecompileAddress};
+use c_kzg::{Bytes32, Bytes48};
+use revm_primitives::{PrecompileResult, StandardPrecompileFn};
+
+pub const POINT_EVALUATION_PRECOMPILE: PrecompileAddress = PrecompileAddress(
+    crate::u64_to_b160(10),
+    Precompile::Standard(point_evaluation_run as StandardPrecompileFn),
+);
+use crate::Error;
+/// `BLS_MODULUS: = 52435875175126190479447740508185965837690552500527637822603658699938581184513`
+/// in big endian format
+const BLS_MODULUS: [u8; 32] = [
+    115, 237, 167, 83, 41, 157, 125, 72, 51, 57, 216, 8, 9, 161, 216, 5, 83, 189, 164, 2, 255, 254,
+    91, 254, 255, 255, 255, 255, 0, 0, 0, 1,
+];
+
+/// See: https://eips.ethereum.org/EIPS/eip-4844
+pub fn point_evaluation_run(input: &[u8], gas_limit: u64) -> PrecompileResult {
+    // The data is encoded as follows: versioned_hash | z | y | commitment | proof | with z and y being padded 32 byte big endian values
+    assert!(input.len() == 192);
+    let cost = 5000;
+    if cost > gas_limit {
+        return Err(Error::OutOfGas);
+    }
+
+    // We can always be sure that these will be 48 bytes so this unwrap should be okay
+    let z = Bytes32::from_bytes(&input[32..64]).unwrap();
+    let y = Bytes32::from_bytes(&input[64..96]).unwrap();
+    let commitment = Bytes48::from_bytes(&input[96..144]).unwrap();
+    let versioned_hash = Bytes48::from_bytes(&input[0..32]).unwrap();
+    let proof = Bytes48::from_bytes(&input[144..192]).unwrap();
+    let kzg_settings = c_kzg::KzgSettings::load_trusted_setup_file(
+        "crates/precompile/src/trusted_setup4.txt".into(),
```

Comment:
Do you need this file in runtime? Where should be placed

Reply:
This is a mock trusted setup from that was used for testing with the EF audited bindings from c-kzg, afaik we do need it at runtime and should replace it with the official trusted set up once the ceremony is finished. I am not sure where it should live so i put it here. Do you think it should go in a different directory.